tests/FSharp.Test.Utilities/ILVerifierModule.fs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+namespace FSharp.Test
+
+open FSharp.Test.Compiler
+open System
+open System.IO
+open TestFramework
+
+[<AutoOpen>]
+module ILVerifierModule =
+    let config = initializeSuite ()
+
+    let private exec (dotnetExe: string) args workingDirectory =
+        let arguments = args |> String.concat " "
+        let exitCode, _output, errors = Commands.executeProcess dotnetExe arguments workingDirectory
+        let errors = errors |> String.concat Environment.NewLine
+        errors, exitCode
+
+    let private verifyPEFileCore peverifierArgs (dllFilePath: string) =
```

Comment:
We are regularily spending time to get rid of `.exe` invocations part of our testing and making our test suites faster and less resource-intensive.

I have realized that `ilverify.exe` is just a very thing wrapper (https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/ILVerify/Program.cs ) over the `ILVerification` library - would that be an option?

https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/ILVerification/Verifier.cs


Once a mechanism around `.exe` goes in, it will be difficult to get rid of it.
(this might be obviously premature if we use it just a handful of times **for now**. But we might to e.g. bulk+async it for all emitted IL and then the cost does add up)

Reply:
Maybe, eventually, but not just now.  And that wouldn't work for the net472 build of the compiler because there is no desktop version of ilverify.